server/monitor.go

```diff
@@ -45,13 +45,14 @@ import (
 
 // Connz represents detailed information on current client connections.
 type Connz struct {
-	ID       string      `json:"server_id"`
-	Now      time.Time   `json:"now"`
-	NumConns int         `json:"num_connections"`
-	Total    int         `json:"total"`
-	Offset   int         `json:"offset"`
-	Limit    int         `json:"limit"`
-	Conns    []*ConnInfo `json:"connections"`
+	ID       string            `json:"server_id"`
+	Now      time.Time         `json:"now"`
+	NumConns int               `json:"num_connections"`
+	Total    int               `json:"total"`
+	Offset   int               `json:"offset"`
+	Limit    int               `json:"limit"`
+	Conns    []*ConnInfo       `json:"connections"`
+	Metadata map[string]string `json:"metadata,omitempty"`
```

Comment:
We probably only need it in `ServerInfo` and `Varz` as `ServerInfo` will go with all the other Z responses already